docs/src/rules/no-await-in-loop.md

```diff
@@ -19,21 +19,53 @@ async function foo(things) {
   const results = [];
   for (const thing of things) {
     // Bad: each loop iteration is delayed until the entire asynchronous operation completes
-    results.push(await bar(thing));
+    results.push(await doAsyncWork(thing));
   }
-  return baz(results);
+  return results;
 }
 ```
 
 ```js
 async function foo(things) {
-  const results = [];
+  const promises = [];
   for (const thing of things) {
     // Good: all asynchronous operations are immediately started.
-    results.push(bar(thing));
+    promises.push(doAsyncWork(thing));
   }
   // Now that all the asynchronous operations are running, here we wait until they all complete.
-  return baz(await Promise.all(results));
+  const results = await Promise.all(promises);
+  return results;
+}
+```
+
+This can be beneficial for subtle error-handling reasons as well. Given an array of promises that might reject,
+sequential awaiting puts the program at risk of unhandled promise rejections. The exact behavior of unhandled
+rejections depends on the environment running your code, but they are generally considered harmful regardless.
+In node.js, for example, [unhandled rejections cause a program to terminate](https://nodejs.org/api/cli.html#--unhandled-rejectionsmode) unless configured otherwise.
+
+```js
+async function foo() {
+    const arrayOfPromises = somethingThatCreatesAnArrayOfPromises();
+    for (const promise of arrayOfPromises) {
+        // Bad: if any of the promises reject, an exception is thrown, and
+        // subsequent loop iterations will not run. Therefore, rejections later
+        // in the array will become unhandled rejections that cannot be caught
+        // by a caller.
+        const value = await promise;
+        console.log(value);
+    }
+}
+```
+
+```js
+async function foo() {
+    const arrayOfPromises = somethingThatCreatesAnArrayOfPromises();
+    // Good: Any rejections will cause a single exception to be thrown here,
+    // which may be caught and handled by the caller.
+    const arrayOfValues = Promise.all(arrayOfPromises);
```

Comment:
I think the intent is to await the promise to throw an exception?

```suggestion
    const arrayOfValues = await Promise.all(arrayOfPromises);
```

Reply:
heh, yep, thanks!